### ISSUE TITLE
Bug when trying to set params for a console command

### ISSUE THREAD

**Issue by brainyl** [yiisoft/yii2#8641]
I get this error that says yii\console\Exception: Unknown option: --param1 in C:\lamp\www\klusster\vendor\yiisoft\yii2\console\Controller.php on line 88. This happens when i supply --param1 as a parameter to the console command.

The error is from the code below, I suspect "(!is_int($name)" is suppose to be "(is_int($name)". Which will imply that numbers are not allowed when setting the paramerers.

``` php
/**
     * Runs an action with the specified action ID and parameters.
     * If the action ID is empty, the method will use [[defaultAction]].
     * @param string $id the ID of the action to be executed.
     * @param array $params the parameters (name-value pairs) to be passed to the action.
     * @return integer the status of the action execution. 0 means normal, other values mean abnormal.
     * @throws InvalidRouteException if the requested action ID cannot be resolved into an action successfully.
     * @throws Exception if there are unknown options or missing arguments
     * @see createAction
     */
    public function runAction($id, $params = [])
    {
        if (!empty($params)) {
            // populate options here so that they are available in beforeAction().
            $options = $this->options($id);
            foreach ($params as $name => $value) {

                if (in_array($name, $options, true)) {
                    $default = $this->$name;
                    $this->$name = is_array($default) ? preg_split('/\s*,\s*/', $value) : $value;
                    unset($params[$name]);
                } elseif (!is_int($name)) {
                    throw new Exception(Yii::t('yii', 'Unknown option: --{name}', ['name' => $name]));
                }
            }
        }
        return parent::runAction($id, $params);
    }
```

**Comment by klimov-paul** [comment]
What is your controller action signature?

**Comment by brainyl** [comment]
public function actionBar($param1, $param2)

**Comment by cebe** [comment]
how exactly do you call the action?

**Comment by cebe** [comment]
See the docs: http://www.yiiframework.com/doc-2.0/guide-tutorial-console.html#arguments

**Comment by brainyl** [comment]
When i use yii foo/bar --param1="test" --param2="testing" it fails but yii foo/bar --0="test" --1="testing" on the other hand works

**Comment by cebe** [comment]
the correct syntax would be `./yii foo/bar test testing`.
The second syntax should fail though...

**Comment by brainyl** [comment]
Okay, I get it now. Thank you!

**Comment by portenko** [comment]
Hi! I've some problem with console in yii2 advanced.
I tried to run console commands:
"php yii cron/test" or "./yii cron/test"
and after that I got the same error "Error: Unknown command cron/test".

I tried to search some information in the internet, but I not found anything.
I create my own console controller and test action by this link http://www.yiiframework.com/doc-2.0/guide-tutorial-console.html#create-command
